deployment/data-streams/changeset/jd_distribute_bootstrap_jobs.go

```diff
@@ -0,0 +1,91 @@
+package changeset
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	chainsel "github.com/smartcontractkit/chain-selectors"
+	"github.com/smartcontractkit/chainlink/deployment"
+	"github.com/smartcontractkit/chainlink/deployment/data-streams/jd"
+	"github.com/smartcontractkit/chainlink/deployment/data-streams/jobs"
+
+	jobv1 "github.com/smartcontractkit/chainlink-protos/job-distributor/v1/job"
+)
+
+const (
+	defaultBootstrapJobSpecsTimeout = 120 * time.Second
+)
+
+var CsDistributeBootstrapJobSpecs deployment.ChangeLogic[CsDistributeBootstrapJobSpecsConfig] = csDistributeBootstrapJobSpecs
+
+type ContractID string
+
+type CsDistributeBootstrapJobSpecsConfig struct {
+	ChainSelectorEVM uint64
+	DONFilter        *jd.DONFilter
+}
+
+func findConfiguratorAddressByDON(addresses map[ContractID]deployment.TypeAndVersion, DONID uint64) (string, error) {
+	for address, contract := range addresses {
+		if contract.Type == "Configurator" && contract.Labels.Contains(fmt.Sprintf("don-%d", DONID)) {
+			return string(address), nil
+		}
+	}
+	return "", fmt.Errorf("Configurator contract not found for DON %d", DONID)
+}
+
+func csDistributeBootstrapJobSpecs(e deployment.Environment, cfg CsDistributeBootstrapJobSpecsConfig) (deployment.ChangesetOutput, error) {
+	ctx, cancel := context.WithTimeout(e.GetContext(), defaultBootstrapJobSpecsTimeout)
+	defer cancel()
+
+	bootstrapNodes, err := jd.FetchDONBootstrappersFromJD(ctx, e.Offchain, cfg.DONFilter)
```

Comment:
Maybe we should validate that there is at least one node.

Reply:
Even better we should check that `len(listedNodes)` matches the expected number of bootstraps for the input DON